pkg/resources/function_commons.go

```diff
@@ -0,0 +1,341 @@
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Optional: true,
+		// TODO [SNOW-1348103]: handle dynamic comment - this is a workaround for now
+		Default:     "user-defined function",
+		Description: "Specifies a comment for the function.",
+	},
+	// TODO [SNOW-1348103]: because of https://docs.snowflake.com/en/sql-reference/sql/create-function#id6, maybe it will be better to split into stage_name + target_path
+	"imports": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: what do we do with the version "latest".
+	"packages": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	"handler": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"external_access_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		},
+		Optional:    true,
+		ForceNew:    true,
+		Description: "The names of [external access integrations](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) needed in order for this function’s handler code to access external networks. An external access integration specifies [network rules](https://docs.snowflake.com/en/sql-reference/sql/create-network-rule) and [secrets](https://docs.snowflake.com/en/sql-reference/sql/create-secret) that specify external locations and credentials (if any) allowed for use by handler code when making requests of an external network, such as an external REST API.",
+	},
+	"secrets": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"secret_variable_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The variable that will be used in handler code when retrieving information from the secret.",
+				},
+				"secret_id": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Fully qualified name of the allowed secret. You will receive an error if you specify a SECRETS value whose secret isn’t also included in an integration specified by the EXTERNAL_ACCESS_INTEGRATIONS parameter.",
+					DiffSuppressFunc: suppressIdentifierQuoting,
+				},
+			},
+		},
+		Description: "Assigns the names of secrets to variables so that you can use the variables to reference the secrets when retrieving information from secrets in handler code. Secrets you specify here must be allowed by the [external access integration](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) specified as a value of this CREATE FUNCTION command’s EXTERNAL_ACCESS_INTEGRATIONS parameter.",
+	},
+	"target_path": {
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+	},
```

Comment:
should it be a string or more structured schema like { stage: string, path: string }? Probably string, because I'm guessing it's returned that way and we don't have to do any parsing which could introduce additional issues

Reply:
this is the same case as TODO left for IMPORTS: `	// TODO [SNOW-1348103]: because of https://docs.snowflake.com/en/sql-reference/sql/create-function#id6, maybe it will be better to split into stage_name + target_path
` - I will add the same TODO for it, and we will discuss tomorrow

Reply:
Added